src/main/java/org/opensearch/timeseries/constant/CommonMessages.java

```diff
@@ -0,0 +1,84 @@
+    public static String CAN_NOT_FIND_RESULT_INDEX = "Can't find result index ";
+    public static String INVALID_CHAR_IN_RESULT_INDEX_NAME =
+        "Result index name has invalid character. Valid characters are a-z, 0-9, -(hyphen) and _(underscore)";
+    public static String FAIL_TO_VALIDATE = "fail to validate";
+    public static String INVALID_TIMESTAMP = "Timestamp field: (%s) must be of type date";
+    public static String NON_EXISTENT_TIMESTAMP = "Timestamp field: (%s) is not found in index mapping";
+    public static String INVALID_NAME = "Valid characters for name are a-z, A-Z, 0-9, -(hyphen), _(underscore) and .(period)";
+    // change this error message to make it compatible with old version's integration(nexus) test
+    public static String FAIL_TO_FIND_CONFIG_MSG = "Can't find config with id: ";
+    public static final String CAN_NOT_CHANGE_CATEGORY_FIELD = "Can't change category field";
+    public static final String CAN_NOT_CHANGE_CUSTOM_RESULT_INDEX = "Can't change custom result index";
+    public static final String CATEGORICAL_FIELD_TYPE_ERR_MSG = "A categorical field must be of type keyword or ip.";
+    // Modifying message for FEATURE below may break the parseADValidationException method of ValidateAnomalyDetectorTransportAction
+    public static final String FEATURE_INVALID_MSG_PREFIX = "Feature has an invalid query";
+    public static final String FEATURE_WITH_EMPTY_DATA_MSG = FEATURE_INVALID_MSG_PREFIX + " returning empty aggregated data: ";
+    public static final String FEATURE_WITH_INVALID_QUERY_MSG = FEATURE_INVALID_MSG_PREFIX + " causing a runtime exception: ";
+    public static final String UNKNOWN_SEARCH_QUERY_EXCEPTION_MSG =
+        "Feature has an unknown exception caught while executing the feature query: ";
+
+    // ======================================
+    // Index message
+    // ======================================
+    public static final String CREATE_INDEX_NOT_ACKNOWLEDGED = "Create index %S not acknowledged";
+    public static final String SUCCESS_SAVING_RESULT_MSG = "Result saved successfully.";
+    public static final String CANNOT_SAVE_RESULT_ERR_MSG = "Cannot save results due to write block.";
+
+    // ======================================
+    // Resource constraints
+    // ======================================
+    public static final String MEMORY_CIRCUIT_BROKEN_ERR_MSG = "AD memory circuit is broken.";
+
+    // ======================================
+    // Transport
+    // ======================================
+    public static final String INVALID_TIMESTAMP_ERR_MSG = "timestamp is invalid";
+
+    // ======================================
+    // transport/restful client
+    // ======================================
+    public static final String WAIT_ERR_MSG = "Exception in waiting for result";
+    public static final String ALL_FEATURES_DISABLED_ERR_MSG =
+        "Having trouble querying data because all of your features have been disabled.";
```

Comment:
shorten to "because all features have been disabled"

Reply:
the msg has been hard coded in the frontend: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/710e3c4ddadf3689b7f014b5df88f2a82b8db999/public/pages/DetectorDetail/utils/constants.ts#L57. Changing it will break frontend. Can we leave it as it is?

Reply:
! yes, let's not break other artifacts.